README.md

```diff
@@ -27,20 +27,21 @@ Metrics are collected from the following components
 
 ## Contributing new component integrations
 
-The Cluster Monitoring Operator has many builtin `ServiceMonitor` resources which enable discovering the metrics endpoints of a variety of well-known components.
+The Cluster Monitoring Operator has many builtin `ServiceMonitor` resources which enable discovering the metrics endpoints of a variety of well-known components. Only components that must be created before the cluster monitoring stack belong in this repository, in order to solve this cyclic dependencies of bootstrapping.
 
 To register a new builtin component, make the following changes:
 
-* Add a new `ServiceMonitor` manifest file to [assets/prometheus-k8s](assets/prometheus-k8s) following the existing `prometheus-k8s-service-monitor-$COMPONENT.yaml` naming convention.
-* Add a constant in [pkg/manifests/manifests.go](pkg/manifests/manifests.go) which points to the new manifest file.
+* Add a new `ServiceMonitor` manifest file to [jsonnet/prometheus.jsonnet](jsonnet/prometheus.jsonnet). An example of this can be seen for the OpenShift component "kube-controllers", [here](https://github.com/openshift/cluster-monitoring-operator/blob/01bfe3789117e7074e893251f2f6d31c816db8fb/jsonnet/prometheus.jsonnet#L113-L145).
+* Re-generate the go-bindata code, using the `pkg/manifests/bindata.go` make target. This will also create a new file in `assets/prometheus-k8s/` according to the name given in the jsonnet code.
+* Add a constant in [pkg/manifests/manifests.go](pkg/manifests/manifests.go) which points to the new manifest file, from `assets/`.
 * Add a new `Factory` method in [pkg/manifests/manifests.go](pkg/manifests/manifests.go) which loads the manifest using the new constant.
 * Add a step to `PrometheusTask` in [pkg/tasks/prometheus.go](pkg/tasks/prometheus.go) which creates the `ServiceMonitor` using the `Factory` new method.
 
-To add a new builtin alerting rule:
+To add a new builtin recording or alerting rules:
```

Comment:
`a` should agree with the number of rules. Either make `rules` singular or remove the `a`